pkg/storage/image.go

```diff
@@ -599,19 +600,30 @@ func GetImageService(ctx context.Context, store storage.Store, defaultTransport
 		seenRegistries[r] = true
 	}
 
+	systemRegistries, err := sysregistriesv2.FindUnqualifiedSearchRegistries(sc)
+	if err != nil {
+		return nil, err
+	}
+
 	is := &imageService{
 		store:                 store,
 		defaultTransport:      defaultTransport,
 		indexConfigs:          make(map[string]*indexInfo),
 		insecureRegistryCIDRs: make([]*net.IPNet, 0),
-		registries:            cleanRegistries,
 		imageCache:            make(map[string]imageCacheItem),
 		ctx:                   ctx,
 	}
+	if len(cleanRegistries) != 0 {
+		is.registries = cleanRegistries
+	} else {
+		for _, r := range systemRegistries {
+			is.registries = append(is.registries, r.URL)
+		}
+	}
 
-	insecureRegistries = append(insecureRegistries, "127.0.0.0/8")
+	crioInsecureRegistries = append(crioInsecureRegistries, "127.0.0.0/8")
```

Comment:
(Absolutely non-blocking: Rename `crioInsecureRegistries` back to `insecureRegistries`, and make the PR and the variable name shorter?)